johnpapa/vscode-peacock
# Improve activity bar styling and introduce color library

## Purpose

This change is to improve the visual appearance and usability of the activity bar when applying a color. 

Additionally, adds [tinycolor](https://github.com/bgrins/TinyColor) to:
* perform color manipulation
* improve flexibility of color input (see [Accepted String Input](https://github.com/bgrins/TinyColor#accepted-string-input))
* enhance color capabilities while reducing extension code base

## What

The default styling of the activity bar in VS Code presents the activity bar in a slightly different shade than the status and title bars to better visually delineate it. It also presents the icons in a more subdued shade when not active or not as the inactive state.

<img width="100" alt="screen shot 2019-03-01 at 1 31 02 pm" src="https://user-images.githubusercontent.com/1085791/53661930-34477480-3c27-11e9-97f0-3655cd43f0e2.png"> 

Currently, Peacock colorizes the activity bar as the same overall background color as the others. It also sets the inactive foreground color of the icons to the same as the active foreground color. Both of these combine to degrade the look and usability of the activity bar when using a Peacock color.

<img width="100" alt="screen shot 2019-03-01 at 1 02 18 pm" src="https://user-images.githubusercontent.com/1085791/53662106-b0da5300-3c27-11e9-8255-539658569d23.png">

This change uses a combination of color modifications to properly stylize the activity bar to better match the default behavior of VS Code so that the switch to Peacock is a bit more seamless.

<img width="100" alt="screen shot 2019-03-01 at 1 03 20 pm" src="https://user-images.githubusercontent.com/1085791/53662271-1e867f00-3c28-11e9-8e0d-f5bb5703fc75.png">

## How to Test

Tests have not been written specific to the new styling, but will be added. All current tests pass following refactoring. To test manually, use Peacock to set random colors or particularly light or dark colors and verify that icons remain visible.

## What to Check

The library allowed for refactoring of the existing color-library code to reduce it. It also manages color input with a much larger array of input that allows for most anything that CSS understands. Existing functionality should be checked, especially the use of named colors since the library replaces some of that functionality.
